### DISCUSSION TITLE
Cortex block storage S3 endpoint issue

### DISCUSSION THREAD

**Discussion by vikram-zopsmart** [cortexproject/cortex#5642] [General]
We have setup cortex in EKS cluster , through HELM deployment mechanism. So in Helm chart we provided

AWS S3 config i.e.

bucketname
endpoint
region
accesskey
secretkey.

Now, the endpoint for the S3 bucket is "s3.us-west-2.amazonaws.com" through Helm deployment values.yaml file.
but When check the cortex compactor pod using "netstat" command it is trying to establish connection to S3 bucket in below format i.e.
"s3-us-west-2-r-w.amazonaws.com".

Since, the endpoint is different then what we have provided, the connection to S3 is going via public IP/NAT gateway

Please suggest any way, if we can make the calls to S3 from compactor through Private IP's.

**Comment by friedrichg** [comment]
This is more an AWS question, but I will answer it: if you add a gateway endpoint to s3 in the VPC, the address s3.us-west-2.amazonaws.com resolves to an internal ip.
https://docs.aws.amazon.com/vpc/latest/privatelink/vpc-endpoints-s3.html
